GameFighter/Models/Lancer.cs

```diff
@@ -0,0 +1,18 @@
+﻿namespace GameFighter.Models
+{
+    public class Lancer : Warrior
+    {
+        public Lancer()
+        {
+            Attack = 6;
+        }
+
+        public override void Attacks(Warrior warrior, Army warriorsArmy, Army thisArmy = null)
+        {
+            var actualAttack = warrior.GetAttack(Attack);
+
+            if (warriorsArmy.Next(warrior) != null)
```

Comment:
You allow to pass null for warriorsArmy parameter. This can lead to NullreferenceException from `Fight(Warrior warrior1, Warrior warrior2, Army army1 = null, Army army2 = null)` method